tests/integration/solidity_contracts/Counter/test_counter.py

```diff
@@ -222,7 +222,144 @@ async def test_extcodecopy_offset_and_size_gte_twenty_c(
-            == expected_memory_result[offset : offset + size]
+        assert memory_result[dest_offset:size+dest_offset] == expected_memory_result[offset:offset+size]
+
+    async def test_extcodecopy_should_pad_zeroes_on_no_account_match(self, deploy_solidity_contract: Callable, kakarot: StarknetContract):
+
+        counter = await deploy_solidity_contract(
+            "Counter", caller_address=1
         )
+        
+        # instructions
+        push1 = 60
+        push20 = 73        
+        extcodecopy = "3c"
+
+        # stack elements
+        offset = 0
+        size = 32
+        dest_offset = 0
+        evm_contract_address = counter.contract_account.deploy_call_info.result.evm_contract_address
+
+        offset1 = 0
+        size1 = 8
+        dest_offset1 = 1
+        evm_contract_address1 = 981189583494067065842568011418895108651581577619
+
+        # we set up instructions to have bytecode in the memory
+        # in this case the bytecode of the deployed counter contract
+        byte_code_match = f"{push1}\
+        {size:02x}\
+        {push1}\
+        {offset:02x}\
+        {push1}\
+        {dest_offset:02x}\
+        {push20}\
+        {evm_contract_address:x}\
+        {extcodecopy}"
+
+        assert evm_contract_address1 != evm_contract_address
+        
+        # then we set up instructions to attempt to copy code from an account that doesn't exist
+        # we set the dest_offset to 1 and length to 8
+        byte_code_empty = f"{push1}\
+        {size1:02x}\
+        {push1}\
+        {offset1:02x}\
+        {push1}\
+        {dest_offset1:02x}\
+        {push20}\
+        {evm_contract_address1:x}\
+        {extcodecopy}"
+        
+        res = await kakarot.execute(
+            value=int(0),
+            bytecode=hex_string_to_bytes_array(byte_code_match+byte_code_empty),
+            calldata=hex_string_to_bytes_array(""),
+        ).call(caller_address=1)
+
+        expected_memory_result = hex_string_to_bytes_array(counter.bytecode.hex())
+        memory_result = extract_memory_from_execute(res.result)
+
+        # we write in zeros at an dest_offset of one, so we expect the contract byte code to still match for the first element
+        assert memory_result[0] == expected_memory_result[0]
+        # we assert that from the dest_offset to the size of the extcodecopy for an address without code, we get zeros
+        assert memory_result[dest_offset1:dest_offset1+size1] == [0] * size1
+        # this assert fails because of discrepancies between locally deployed contract and compiled.
+        # will investigate, but we would the assert that the rest of memory matches with the copied bytecode of the first contract
+        # where 'rest' is the end of the zero'ed out memory dest_offset+size1+1
+        end_of_zero_padded_region = dest_offset+size1+1
+        # assert memory_result[end_of_zero_padded_region:size] == expected_memory_result[end_of_zero_padded_region:size]
```

Comment:
so the issue is not solved yet?

Reply:
I believe that this is due to https://github.com/sayajin-labs/kakarot/issues/342, hence why there are still skipped here.